api/src/lib.rs

```diff
@@ -39,9 +39,8 @@ fn read_nullable_raw_pointer<'a, T>(input: *const T) -> Option<&'a T> {
 fn serialize_from_raw_pointer<T: CanonicalSerialize>(
     to_write: *const T,
     compressed: Option<bool>,
-) -> Vec<u8> {
+) -> Result<Vec<u8>, SerializationError> {
     serialize_to_buffer(read_raw_pointer(to_write), compressed)
-        .expect(format!("unable to write {} to buffer", type_name::<T>()).as_str())
```

Comment:
That's a pretty big change actually affecting also the SDK I assume. This function is used by "serialize_from_jobject" function that is used in a lot of places. So the caller should check each time that the return value is not NULL. 
Cleanest way to handle this is through the usage of Java exceptions but we would still need to add lot of throws to each function argument.
Since we don't have much time now, my suggestion would be to leave the .expect() here and put a TODO to handle this more carefully.
@i-Alex what do you think ?

Reply:
I agree with Daniele to leave it as it was before.
There are a lot of places where we need to update the code (JNI part) to cover the "null cases".

Reply:
Ok, I've returned back the previous state and put TODO regarding Java exceptions for future